electrum/wallet.py

```diff
@@ -1323,7 +1326,46 @@ def is_frozen_address(self, addr: str) -> bool:
 
     def is_frozen_coin(self, utxo: PartialTxInput) -> bool:
         prevout_str = utxo.prevout.to_str()
-        return prevout_str in self._frozen_coins
+        frozen = self._frozen_coins.get(prevout_str, None)
+        # note: there are three possible states for 'frozen':
+        #       True/False if the user explicitly set it,
+        #       None otherwise
+        if frozen is None:
+            return self._is_coin_small_and_untrusted(utxo)
+        return bool(frozen)
+
+    def _is_coin_small_and_untrusted(self, utxo: PartialTxInput) -> bool:
```

Comment:
I guess you meant small_and_unconfirmed

Reply:
I simply was not sure how to call it; either is fine.